orderbook/src/api/order_validation.rs

```diff
@@ -0,0 +1,734 @@
+use crate::{account_balances::BalanceFetching, fee::MinFeeCalculating};
+use contracts::WETH9;
+use ethcontract::{H160, U256};
+use model::order::{BuyTokenDestination, Order, SellTokenSource, BUY_ETH_ADDRESS};
+use shared::{bad_token::BadTokenDetecting, web3_traits::CodeFetching};
+use std::{sync::Arc, time::Duration};
+use warp::{http::StatusCode, reply::Json};
+
+pub trait WarpReplyConverting {
+    fn to_warp_reply(&self) -> (Json, StatusCode);
+}
```

Comment:
While I think the trait here is cool, I don't see any generic code with something like `T: WarpReplayConverting` and feels a bit of overkill.

Reply:
I think this will become more useful once #1176 comes into play (so I can implement this trait for a struct that is not declared inside this crate). Would that be a good reason for it?

Reply:
Personally, I would still not use a trait in that situation (for mostly stylistic reasons), but there isn't anything wrong with doing so feel free to leave it in if you like it. I don't see a strong reason for using or not using a trait either way.

I think a more concrete reason to introduce such a trait would be the ability to get rid of all those `*_response` (for example `create_order_response`) methods and introduce a generic one:
```
pub fn as_response<T, E>(result: Result<T, E>) -> impl Reply
where
    T: serde::Serialize,
    E: WarpReplyConverting,
{
    // ...
}
```

Reply:
(Not suggesting this be done in this PR).

Reply:
Cool I will defs do this (after).

Reply:
For now I will move the trait to `api.rs` (i.e. before merging this).